stdlib/SparseArrays/src/higherorderfns.jl

```diff
@@ -1154,4 +1155,58 @@ map(f::Tf, A::SparseOrStructuredMatrix, Bs::Vararg{SparseOrStructuredMatrix,N})
 map!(f::Tf, C::AbstractSparseMatrixCSC, A::SparseOrStructuredMatrix, Bs::Vararg{SparseOrStructuredMatrix,N}) where {Tf,N} =
     (_checksameshape(C, A, Bs...); _noshapecheck_map!(f, C, _sparsifystructured(A), map(_sparsifystructured, Bs)...))
 
+
+# (13) extrema methods optimized for sparse vectors/matrices.
+function _extrema_itr(f, A::SparseVecOrMat)
+    M = length(A)
+    iszero(M) && throw(ArgumentError("Sparse array must have at least one element."))
+    N = nnz(A)
+    iszero(N) && return f(zero(eltype(A))), f(zero(eltype(A)))
+    vmin, vmax = _extrema_itr(f, nonzeros(A))
+    if N != M
+        f0 = f(zero(eltype(A)))
+        vmin = min(f0, vmin)
+        vmax = max(f0, vmax)
+    end
+    vmin, vmax
+end
+
+function _extrema_dims(f, x::SparseVector{Tv, Ti}, dims) where {Tv, Ti}
+    sz = zeros(1)
+    for d in dims
+        sz[d] = 1
+    end
+    if sz == [1] && !iszero(length(x))
+        return [_extrema_itr(f, x)]
+    end
+    invoke(_extrema_dims, Tuple{Any, AbstractArray, Any}, f, x, dims)
+end
+
+function _extrema_dims(f, A::AbstractSparseMatrix{Tv, Ti}, dims) where {Tv, Ti}
+    sz = zeros(2)
+    for d in dims
+        sz[d] = 1
+    end
+    if sz == [1, 0] && !iszero(length(A))
+        B = Array{Tuple{Tv,Tv}}(undef, 1, size(A, 2))
+        @inbounds for col_idx in 1:size(A, 2)
+            col = @view A[:,col_idx]
+            fx = (nnz(col) == size(A, 1)) ? f(A[1,col_idx]) : f(zero(Tv))
+            B[col_idx] = (fx, fx)
+            for x in nonzeros(col)
+                fx = f(x)
+                if fx < B[col_idx][1]
+                    B[col_idx] = (fx, B[col_idx][2])
+                elseif fx > B[col_idx][2]
+                    B[col_idx] = (B[col_idx][1], fx)
+                end
+            end
+        end
+        return B
+    end
+    invoke(_extrema_dims, Tuple{Any, AbstractArray, Any}, f, A, dims)
+end
+
+_extrema_dims(f, A::SparseVecOrMat, ::Colon) = _extrema_itr(f, A)
```

Comment:
I'm not sure, but maybe splitting this into two helps with the ambiguity?
```suggestion
_extrema_dims(f, A::AbstractSparseVector, ::Colon) = _extrema_itr(f, A)
_extrema_dims(f, A::AbstractSparseMatrix, ::Colon) = _extrema_itr(f, A)
```
I believe you could generalize the method on line 1174 towards `AbstractSparseVector`... well, or restrict the matrix methods to `SparseMatrixCSC`. I'm not sure what the API for the abstract ones is.

Reply:
Unfortunately it doesn't work.
The method on line 1174 can't be generalized because nnz is not defined for `AbstractSparseVector`. Almost every function in `Sparsearrays` is defined for `SparseVector` anf `AbstractSparseMatric`, actually.

Reply:
No problem, that's fine. But does splitting the methods into `SparseVector` and `AbstractSparseMatrix` help? I think there might be an ambiguity because the method here is more abstract in the second argument but more concrete in the third (but I forgot compared to which other method).